openmls/src/group/mls_group/creation.rs

```diff
@@ -507,3 +536,69 @@ fn keys_for_welcome<Provider: OpenMlsProvider>(
     }
     Ok((resumption_psk_store, key_package_bundle))
 }
+
+/// Verify or skip the validation of leaf node lifetimes in the ratchet tree
+/// when joining a group.
+#[derive(Default, Clone, Copy, Debug, PartialEq, Eq, Serialize, Deserialize)]
+pub enum LeafNodeLifetimePolicy {
+    /// Verify the lifetime of leaf nodes in the ratchet tree.
+    ///
+    /// **NOTE:** Only leaf nodes that have never been updated have a lifetime.
+    #[default]
+    Verify,
+
+    /// Skip the verification of the lifeimte in leaf nodes in the ratchet tree.
+    Skip,
+}
+
+/// Builder for joining a group.
+///
+/// Create this with [`StagedWelcome::build_from_welcome`].
+pub struct JoinBuilder<'a, Provider: OpenMlsProvider> {
+    provider: &'a Provider,
+    processed_welcome: ProcessedWelcome,
+    ratchet_tree: Option<RatchetTreeIn>,
+    validate_lifetimes: LeafNodeLifetimePolicy,
+}
+
+impl<'a, Provider: OpenMlsProvider> JoinBuilder<'a, Provider> {
+    /// Create a new builder for the [`JoinBuilder`].
+    pub(crate) fn new(provider: &'a Provider, processed_welcome: ProcessedWelcome) -> Self {
+        Self {
+            provider,
+            processed_welcome,
+            ratchet_tree: None,
+            validate_lifetimes: LeafNodeLifetimePolicy::Verify,
+        }
+    }
+
+    /// The ratchet tree to use for the new group.
+    pub fn with_ratchet_tree(mut self, ratchet_tree: RatchetTreeIn) -> Self {
+        self.ratchet_tree = Some(ratchet_tree);
+        self
+    }
+
+    /// Skip the validation of lifetimes in leaf nodes in the ratchet tree.
+    ///
+    /// Note that only the leaf nodes are checked that were never updated.
+    pub fn skip_lifetime_validation(mut self) -> Self {
+        self.validate_lifetimes = LeafNodeLifetimePolicy::Skip;
+        self
+    }
```

Comment:
The ExternalCommitBuilder and the JoinBuilder have different interfaces w.r.t. lifetime validation policy. Maybe it would be worth aligning the two?